### PR TITLE
Login functionality

### PR DESCRIPTION
**Pull request by GoldenCherro**
i added the login related files and changes, CORS in the api and a command to run both Front and Back at the same time

### CODE REVIEW

File: cherro-front/package.json [hunk 1]
```diff
@@ -3,14 +3,18 @@
...
     "axios": "^0.19.2",
+    "bootstrap": "^4.5.3",
     "enzyme": "^3.11.0",
     "enzyme-adapter-react-16": "^1.15.5",
     "eslint-config-airbnb": "^18.2.1",
     "history": "^4.10.1",
     "node-sass": "^4.14.1",
     "prop-types": "^15.7.2",
     "react": "^16.13.1",
+    "react-bootstrap": "^1.4.0",
```
Comment (mlvieras) [inline cherro-front/package.json:18]: Are you using react-bootstrap?

File: cherro-front/src/models/user.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*
+  NOTE: this file serves only as an example and is not used.
```
Comment (mlvieras) [inline cherro-front/src/models/user.js]: We don't need this comment here

File: cherro-front/src/networking/controllers/login-controller.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  You can remove it altogether or repurpose it by changing its name.
+  Learn more about our networking architecture on:
+  https://blog.xmartlabs.com/2020/07/09/frontend-architecture-and-best-practices/
+*/
+class LoginController {
+  static async logIn(email, password) {
+    try {
+      const response = await ApiService.post(API_ROUTES.LOGIN,
+        { user: { email, password } });
+      return response.data;
```
Comment (mlvieras) [inline cherro-front/src/networking/controllers/login-controller.js]: You need to deserialize data and create models here

File: cherro-front/src/networking/controllers/login-controller.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { ApiService } from '../api-service';
+import { API_ROUTES } from '../api-routes';
+
+/*
+  NOTE: this file serves only as an example and is not used.
```
Comment (mlvieras) [inline cherro-front/src/networking/controllers/login-controller.js]: This comment isn't needed here

File: cherro-front/src/pages/login/login.jsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
+import Alert from 'react-bootstrap/Alert';
+// import InputGroup from 'react-bootstrap/InputGroup';
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.jsx]: Commented code

File: cherro-front/src/pages/login/login.jsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+import Button from '@material-ui/core/Button';
+import { TextField } from '@material-ui/core';
+import {
+  AppLink,
+  goToPage,
+  routeNaming,
+} from '../../routes';
+import styles from './login.module.scss';
+// import globalStyles from '../../assets/stylesheets/global-styles.module.scss';
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.jsx]: Here too

File: cherro-front/src/pages/login/login.jsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+import { LoginController } from '../../networking/controllers/login-controller';
+
+const Login = () => {
+  const [nickname, setNickname] = useState('');
+  const [password, setPassword] = useState('');
+  let msg = '';
+
+  const logIn = async (data) => {
+    data.preventDefault();
+    const loggear = await LoginController.logIn(nickname, password);
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.jsx]: Careful with spanish 😄

File: cherro-front/src/pages/login/login.jsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        </Grid>
+        <Grid
+          container
+          className={styles.logInGrid}
+          direction="row"
+          justify="center"
+          alignItems="center"
+        >
+          <Button
+            id="logInButton"
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.jsx]: Please avoid using IDs unless necessary

File: cherro-front/package.json (hunk 1 above)
Comment (GoldenCherro) [reply cherro-front/package.json:17]: yes, im using it to show an Alert in the login page

File: cherro-front/package.json [hunk 9]
```diff
@@ -7,17 +7,18 @@
...
     "history": "^4.10.1",
     "node-sass": "^4.14.1",
     "prop-types": "^15.7.2",
     "react": "^16.13.1",
     "react-bootstrap": "^1.4.0",
     "react-dom": "^16.13.1",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "4.0.1"
+    "react-scripts": "4.0.1",
+    "enzyme": "^3.11.0",
```
Comment (mlvieras) [inline cherro-front/package.json]: Can we try to have the packages sorted by name? 😬

File: cherro-front/src/pages/login/login.test.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-unused-vars */
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.test.js]: Why is this disabled?

File: cherro-front/src/pages/login/login.test.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+jest.mock('../../networking/controllers/login-controller');
+
+const faker = require('faker');
+
+describe('Login', () => {
+  const setupTest = async (nick, pass, output) => {
+    const nickField = output.find({ label: 'Nickname' }).first();
+    const passField = output.find({ label: 'Password' }).first();
+
+    await act(async () => {
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.test.js]: Why is this method async?

File: cherro-front/src/pages/login/login.test.js [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    return output;
+  };
+
+  it('should render correctly', () => {
+    const output = shallow(<Login />);
+    expect(output.exists()).toBe(true);
+  });
+
+  describe('when nickname and password are entered', () => {
+    it('login is enabled', async () => {
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.test.js]: Phrasing of test descriptors is important to allow easier reading of test suites. You can read more [here](https://www.bignerdranch.com/blog/why-do-javascript-test-frameworks-use-describe-and-beforeeach/)

```suggestion
    it('has an enabled submit button', async () => {
```

This comment also applies to other `it` calls 😄

File: cherro-front/src/pages/login/login.test.js [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+      const output = shallow(<Login />);
+      setupTest('', '', output);
+
+      const logInButtonD = output.find({ type: 'submit' }).first();
+      expect(logInButtonD.props().disabled).toBe(true);
+    });
+  });
+
+  describe('Loggin In', () => {
+    const setupLogginIn = async (nick, pass, obj) => {
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.test.js]:
```suggestion
    const setupLoggingIn = async (nick, pass, obj) => {
```

File: cherro-front/src/pages/login/login.test.js [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+      const loginForm = output.find('form').first();
+
+      LoginController.logIn.mockImplementationOnce(() => (obj));
+
+      await act(async () => {
+        loginForm.props().onSubmit({ preventDefault: () => {} });
+      });
+    };
+
+    it('calls controller with nickname and password', async () => {
```
Comment (mlvieras) [inline cherro-front/src/pages/login/login.test.js:67]: Another good case to test is that the `logIn` method is called only once

**Review by mlvieras — APPROVED** [review]
👏

**Pull request merged**